packages/fxa-content-server/app/scripts/lib/constants.js

```diff
@@ -162,4 +162,8 @@ module.exports = {
   // This is compared against all secondary email
   // records, both verified and unverified
   MAX_SECONDARY_EMAILS: 3,
+
+  // Allow ID Tokens used as the id_token_hint argument in a prompt=none
+  // request to be this many seconds past their expiration.
+  ID_TOKEN_HINT_GRACE_PERIOD: 60 * 60 * 24 * 7,
```

Comment:
Was this default specified in the spec? A weeks seems really long to me.

Reply:
I didn't see any discussion of specific times in the spec. The alternative here is for RPs to pass the user's email as a `login_hint`, and we do allow that for a controlled set of RPs, so it seemed reasonable to be liberal with the staleness of the token used as the `id_token_hint`. @rfk, any thoughts on a good upper limit?

Reply:
I don't think there's any harm in accepting an arbitrary lifetime here. If I signed in to AMO, went away for two months, came back to AMO and had to re-authenticate, it seems entirely reasonable for AMO to send through the two-month-old id_token and for us to accept it as a hint for which user to sign in.

I don't think we're making any security decisions based on the data in this token (yet), are we? We're using it for the single bit of information of "has the user signed in to this RP in the past?".

Reply:
FWIW AMO doesn't store these tokens. We ask the `/v1/token` endpoint to give us one from `code` + `client_id` + `client_secret` when the user comes back from FxA at login (thereby verifying that the `code` is correct), and for the case we want to force 2FA on that user, immediately redirect the user back to FxA with the token we just got as `id_token_hint` so that they don't have to re-authenticate. So if `/v1/token` gives us fresh tokens, we have no reason to ever submit an old one.

Reply:
@rfk That was my read of the spec as well, thanks for weighing in. Probably will leave it as-is for now, I guess 🤷‍♂️ 

> I don't think we're making any security decisions based on the data in this token (yet), are we? We're using it for the single bit of information of "has the user signed in to this RP in the past?".

Yup. We don't do anything with this token beyond verify it's valid and check that, if there's currently a logged-in user, it matches the account in the token.